src/MongoDB.Driver/PipelineStageDefinitionBuilder.cs

```diff
@@ -1429,15 +1429,23 @@ public static class PipelineStageDefinitionBuilder
         }
 
         // private methods
-        private static bool IsTConnectToOrEnumerableTConnectTo<TConnectFrom, TConnectTo>()
+        private static bool AreGraphLookupTypesCompatible<TConnectFrom, TConnectTo>()
```

Comment:
Nit: Didn't we agree on `AreGraphLookupFromAndToTypesCompatible` as the name?

There are more types involved in GraphLookup than just From and To. This method does not check that ALL types involved in GraphLookup are compatible, only the From and To types.